src/filesystem/interface.js

```diff
@@ -206,6 +206,48 @@ function FileSystem(options, callback) {
     return watcher;
   };
 
+  //Object that uses filenames as keys
+  const statWatchers = new Map(); 
+
+  this.watchFile = function(filename, options, listener) {
+    const prevStat, currStat; 
+
+    if (Path.isNull(filename)) {
+      throw new Error('Path must be a string without null bytes.');
+    }
+    //Checks to see if there were options passed in and if not, the callback function will be set here
+    if (typeof options === 'function') {
+      listener = options; 
+      options = {};
+    }
+    //default 5007ms interval, persistent is not used this project
+    const interval = options.interval || 5007;
+    listener = listener || nop;
+    
+    //Stores initial prev value to compare 
+    fs.stat(filename, function(err, stats) { prevStat = stats}); 
+
+    //stores interval return values
+    statWatchers.set(filename, value);
+
+    var value = setInterval(function() {
+      fs.stat(filename, function(err, stats) {
+        if(err) {
+          console.log(err);
+        }
+        //Store the current stat
+        currStat = stats;
```

Comment:
I thought the order makes sense in this case because initial prevStat and currStat are undefined, so I set an initial stat value to prevStat. In which I then set the stats coming from fs.stat to currStat since it's undefined. In which does a compare and then sets current to prev.

Reply:
Could be.  Let's circle back to this when you get deeper into it.